tsfc/driver.py

```diff
@@ -145,6 +145,18 @@ def compile_integral(integral_data, form_data, prefix, parameters,
         quadrature_degree = params.get("quadrature_degree",
                                        params["estimated_polynomial_degree"])
         try:
+            quadrature_degree = params["quadrature_degree"]
+        except KeyError:
+            quadrature_degree = params["estimated_polynomial_degree"]
+            functions = list(arguments) + [coordinates] + list(integral_data.integral_coefficients)
+            function_degrees = [f.ufl_function_space().ufl_element().degree() for f in functions]
+            if all(quadrature_degree > 10 * deg for deg in function_degrees):
```

Comment:
It does not seem to fail the Firedrake test suite for me, but that's probably because in Python 2 `(4, 4) > 10 * (2,2) => (4, 4) > (2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2) => True`.